examples/circuit_random/circuit_random.cc

```diff
@@ -0,0 +1,198 @@
+                cout << 0 << " -> " << s(x[0]);
+                auto current = s(x[0]);
+                while (current != 0_i) {
+                    cout << " -> ";
+                    cout << s(x[current.raw_value]);
+                    current = s(x[current.raw_value]);
+                }
+                cout << "\nMax leg = " << s(max_leg) << endl;
+                cout << "\n";
+
+                return true;
+            },
+        },
+        make_optional<ProofOptions>("circuit_random.opb", "circuit_random.veripb"));
+    return stats;
+}
+
+auto main(int argc, char * argv[]) -> int
+{
+    po::options_description display_options{"Program options"};
+    display_options.add_options()            //
+        ("help", "Display help information") //
+        ("prove", "Create a proof");         //
+
+    po::options_description all_options{"All options"};
+
+    all_options.add_options()(
+        "n", po::value<int>(), "Integer value n.");
+
+    all_options.add_options()(
+        "seed", po::value<unsigned int>(), "Random seed.");
+
+    all_options.add(display_options);
+    po::variables_map options_vars;
+
+    try {
+        po::store(po::command_line_parser(argc, argv)
+                      .options(all_options)
+                      .run(),
+            options_vars);
+        po::notify(options_vars);
+    }
+    catch (const po::error & e) {
+        cerr << "Error: " << e.what() << endl;
+        cerr << "Try " << argv[0] << " --help" << endl;
+        return EXIT_FAILURE;
+    }
+
+    if (options_vars.contains("help")) {
+        cout << "Usage: " << argv[0] << " [options] [size]" << endl;
+        cout << endl;
+        cout << display_options << endl;
+        return EXIT_SUCCESS;
+    }
+
+    if (options_vars.contains("seed")) {
+        if (! options_vars.contains("n")) {
+            cout << "Need to specify n.";
+            return EXIT_FAILURE;
+        }
+        auto n = options_vars["n"].as<int>();
+        auto [distances, seed] = create_graph_from_seed(n, EDGE_PROBABILITY, options_vars["seed"].as<unsigned int>());
+        for (unsigned int i = 0; i < distances.size(); ++i) {
+            cout << i << ": ";
+            for (unsigned int j = 0; j < distances[i].size(); ++j) {
+                if (distances[i][j] != -1)
+                    cout << j << " ";
+            }
+            cout << endl;
+        }
+
+        run_circuit_problem(n, distances);
+    }
+    else {
+        int smallest_n = 8;
+        int largest_n = 20;
+        int repetitions = 20;
+
+        for (int n = smallest_n; n <= largest_n; ++n) {
+            for (int r = 0; r < repetitions; ++r) {
+                auto [distances, seed] = generate_random_graph(n, EDGE_PROBABILITY);
+
+                Stats stats = run_circuit_problem(n, distances);
+
+                cout << "Num solutions: " << stats.solutions << endl;
+
+                if (0 != system("veripb circuit_random.opb circuit_random.veripb")) {
```

Comment:
I'd leave this out, in favour of using the CMake test runner to check the proofs.